tests/imfile-file-descriptor-exhasution.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# This is part of the rsyslog testbench, licensed under GPLv3
+. ${srcdir:=.}/diag.sh init
+. $srcdir/diag.sh check-inotify
+
+generate_conf
+add_conf '
+# comment out if you need more debug info:
+	global( debug.whitelist="on"
+		debug.files=["imfile.c"])
+
+module(load="../plugins/imfile/.libs/imfile"
+       mode="inotify" normalizePath="off")
+
+input(type="imfile" File="./'$RSYSLOG_DYNNAME'.input.*.log" Tag="file:"
+	Severity="info" Facility="local7")
+
+'
+# generate input files first. Note that rsyslog processes it as
+# soon as it start up (so the file should exist at that point).
+# creating 1024 files to be parsed by imfile
+for  (( i= 0 ; i < 100 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.log;
+done
+
+# Start rsyslog now
+startup
+
+#keep a boolean to track if FD exhasuted
+
+# keep adding files and check wether the FD exhasuted
+
+for (( i = 100; i < 1200 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.log;
+	if content_check --check-only "Too many open files" $RSYSLOG_DYNNAME.started; then
+		printf 'Failed to open file $RSYSLOG_DYNNAME.input.%d.log a FD exhasuted\n' "$i"
+		break
+	fi
+done
+
+#sleeping for 3 sec, as all FDs are exhasuted and no 
+#listner port available as well incomming connection
```

Comment:
I admit I do not understand why the sleep **here at this point** does help. Can you elaborate?

Reply:
Sleep is not as such required here, i kept it to check even if all FDs exhausted, rsyslog process is not crashing. Let me know your comments as well.

Reply:
if we do not need `sleep` we should remove it. This has two advantages

1. testbench runs quicker (is already quite long running and, yes, single tests sum up)
2. `sleep` inside tests always cause doubt of raciness and thus make devs question why it is there ever and ever again. In sum, this wastes a lot of time.

Reply:
Thanks for your comment. I have removed sleep from the change.